src/libmongoc/src/mongoc/mongoc-rpc.c

```diff
@@ -635,42 +635,46 @@ _mongoc_rpc_gather (mongoc_rpc_t *rpc, mongoc_array_t *array)
 void
 _mongoc_rpc_swab_to_le (mongoc_rpc_t *rpc)
 {
-#if BSON_BYTE_ORDER != BSON_LITTLE_ENDIAN
-   mongoc_opcode_t opcode;
+   BSON_UNUSED (rpc);
 
-   opcode = rpc->header.opcode;
-
-   switch (opcode) {
-   case MONGOC_OPCODE_REPLY:
-      _mongoc_rpc_swab_to_le_reply (&rpc->reply);
-      break;
-   case MONGOC_OPCODE_MSG:
-      _mongoc_rpc_swab_to_le_msg (&rpc->msg);
-      break;
-   case MONGOC_OPCODE_UPDATE:
-      _mongoc_rpc_swab_to_le_update (&rpc->update);
-      break;
-   case MONGOC_OPCODE_INSERT:
-      _mongoc_rpc_swab_to_le_insert (&rpc->insert);
-      break;
-   case MONGOC_OPCODE_QUERY:
-      _mongoc_rpc_swab_to_le_query (&rpc->query);
-      break;
-   case MONGOC_OPCODE_GET_MORE:
-      _mongoc_rpc_swab_to_le_get_more (&rpc->get_more);
-      break;
-   case MONGOC_OPCODE_DELETE:
-      _mongoc_rpc_swab_to_le_delete (&rpc->delete_);
-      break;
-   case MONGOC_OPCODE_KILL_CURSORS:
-      _mongoc_rpc_swab_to_le_kill_cursors (&rpc->kill_cursors);
-      break;
-   case MONGOC_OPCODE_COMPRESSED:
-      _mongoc_rpc_swab_to_le_compressed (&rpc->compressed);
-      break;
-   default:
-      MONGOC_WARNING ("Unknown rpc type: 0x%08x", opcode);
-      break;
+#if BSON_BYTE_ORDER != BSON_LITTLE_ENDIAN
```

Comment:
Aside: I'm strongly in favor of using language-level checks rather than conditional compilation if at all possible. That would remove the need to annotate "this variable might be unused depending on the preprocessor state." It also ensures we have valid code in the conditional block even on systems where the block would otherwise be unevaluated.